shell/platform/windows/fixtures/main.dart

```diff
@@ -86,6 +87,42 @@ void alertPlatformChannel() async {
   ui.PlatformDispatcher.instance.sendPlatformMessage('flutter/accessibility', byteData, (ByteData? _){});
 }
 
+@pragma('vm:entry-point')
+void exitTestExit() async {
+  final Completer<ByteData?> closed = Completer<ByteData?>();
+  ui.channelBuffers.setListener('flutter/platform', (ByteData? data, ui.PlatformMessageResponseCallback callback) async {
+    const String jsonString = '[{"response":"exit"}]';
+    ByteData responseData = ByteData.sublistView(Uint8List.fromList(utf8.encode(jsonString)));
+    callback(responseData);
+    closed.complete(data);
+  });
+  await closed.future;
+
+  // From here down, nothing should be called, because the application will have already been closed.
+  final Completer<ByteData?> exited = Completer<ByteData?>();
+  const String jsonString = '{"method":"System.exitApplication","args":{"type":"required","exitCode":0}}';
```

Comment:
Similarly here, using `json.encode` to encode a map would allow the map to be laid out the way it's structured:
```dart
final String jsonString = json.encode(<String, dynamic>{
    "method": "System.exitApplication",
    "args": <String, dynamic>{
      "type": "required",
      "exitCode": 0,
    }
  };